Cargo.toml

```diff
@@ -19,7 +19,7 @@ default-target = "x86_64-pc-windows-msvc"
 targets = ["aarch64-pc-windows-msvc", "i686-pc-windows-msvc", "x86_64-pc-windows-msvc"]
 
 [dependencies]
-socket2 = "0.3.16"
+socket2 = "0.4.0"
```

Comment:
As far as I can tell, this is only used in the tests. I think we can move this to `[dev-dependencies]`? We then have to remove `extern crate socket2` from `lib.rs`. But I don't understand what that declaration do there at all. This is a Rust 2018 crate, and it's not publicly re-exported either. Am I missing something? :thinking:

Reply:
Likely it's due to the age of the code, written pre Rust 2018. I can remove them if you want.

Reply:
I looked at the open PRs from before I became a maintainer of this crate. There was already a PR on moving `socket2` to a dev-dependency. So I merged that one just now. But it's still not upgraded. So could you please rebase on latest master and still do the upgrade to `0.4`?

Reply:
Done.